README.md

```diff
@@ -1,6 +1,160 @@
-# Anthos Service Mesh packages
+# ASM MCP
+
+This repository contains a blueprint repository for installing the ASM MCP
+client-side components in an ASM-MCP compliant cluster and GCP project.
+
+The tools `kpt` and `anthoscli` are required in order to install the full sample.
+If you intend to install ASM-MCP in an existing cluster, you will also need 
+`kubectl`.
+
+There are three subdirectories, each containing a portion of the installation.
+
+- The `project` subfolder, which contains resources that configure GCP Project
+level resources such as IAM roles and enabled APIs.
+
+- The `cluster` subfolder, which contains resources that create and deploy an
+ASM-MCP compatible GKE cluster.
+
+- The `asm` subfolder, which contains the ASM-MCP cluster-side components.
+At this time, this includes the following:
+  - Network Endpoint Group controller - used to create GCP NEGs for cluster 
+    services. This will eventually be removed as its functionality is replaced.
+  - `asm-galley-push` - Used to push cluster state to the MCP.
+  - `asm-td` - Used to inject sidecar configurations into pods.
+  - Various RBAC resources, configmaps, secrets, and other plumbing to support
+    the main deployments.
+  - A post-install script used to configure the few resources which cannot be
+    managed declaratively (which will be removed in the future)
+  
+# Installation
+
+There are two methods for installing ASM MCP: as a greenfield GKE cluster, or as
+a component in an existing cluster.
+
+## Installing a new cluster
+
+To install a new cluster, you must do the following:
+
+```bash
+#################################
+# Install anthoscli and kpt
+#################################
+gcloud components install kpt
+gcloud components install anthoscli (??) # TODO: Not sure where this comes from now
+
+#################################
+# Configure environment variables
+#################################
+export PROJECT_ID=${the project id}
+export CLUSTER_NAME=my-cluster-name
+# LOCATION can be a zone (for single zone clusters) or region (for regional)
+export LOCATION=us-west1-b
+
+#################################
+# Fetch and configure blueprints
+#################################
+mkdir some/cluster/directory
+cd some/cluster/directory
+# To take full advantage of kpt, this should be a git repo. Run `git-init` or `git clone`
+git init .
+
+kpt pkg get https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages.git/asm@asm-networking asm
+kpt pkg get https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages.git/project@asm-networking project
+kpt pkg get https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages.git/cluster@asm-networking cluster
+
+# To set tunables, use `kpt cfg set`. To see all tunables, run `kpt cfg list-setters <folder>`
+kpt cfg set asm location ${LOCATION}
+kpt cfg set asm cluster-name ${CLUSTER_NAME}
```

Comment:
We've changed names of some setters in 1.5. So also need to update them here to be consistent: 


Set the cluster name:

`kpt cfg set asm gcloud.container.cluster ${CLUSTER_NAME}`

Set the default zone or region:

`kpt cfg set asm gcloud.compute.location ${CLUSTER_LOCATION}`

Set the max node number:

`kpt cfg set asm gcloud.container.nodepool.max-nodes ${MAX_NODES}`